src/include/oshmpi_impl.h

```diff
@@ -184,6 +194,9 @@ typedef struct {
     MPI_Comm comm_world;        /* duplicate of COMM_WORLD */
     MPI_Group comm_world_group;
 
+    OSHMPI_team_t *team_world;
+    OSHMPI_team_t *team_shared;
+
```

Comment:
We certainly can cache those info. I was thinking about you comment on "remove `OSHMPI_global.comm_world` and use `OSHMPI_global.team_world->comm`". I guess using `OSHMPI_global.comm_world` or `OSHMPI_global.team_world_comm` instead of the OSHMPI_global.team_world->comm` would be the preferable choice here.

I think caching them is fine. It would also enables us to create fast-paths in some "team" functions.